sda-commons-server-morphia/README.md

```diff
@@ -91,6 +91,7 @@ The database connection is configured in the `config.yaml` of the application.
 Example config for **production** to be used with environment variables of the cluster configuration:
 ```yaml
 mongo:
+  connectionString: "${MONGODB_CONNECTION_STRING}"
```

Comment:
It may not be suitable to require connectionString and hosts in the `config.yml` any more. We may add validation annotations in `MongoConfiguration` so that one of both is defined. But that would be breaking, if anybody already uses the class in the AppConfiguration with `@Valid`. The easiest would be to suggest an empty default value for connection string and hosts in the `config.yml`.

Reply:
Wouldn't it be possible to stay backwards compatible here?
At the moment `hosts` and `database` are mandatory. I can add some custom validation "connectionString" and the other properties may not be set at the same time.
Why would it break things for clients?

Reply:
What option should we recommend in the documentation? Maybe it would be better to document two separate ways for the configuration: 1) Recommended option (connectionString?), 2) Legacy configuration

Reply:
It's not a backward compatibility issue with sda-dropwizard-commons for the services. But if services change there `config.yml` as suggested - without a default for hosts and connection string - it will be a breaking change for the deployments. When upgrading the service, the operations team must set `MONGODB_CONNECTION_STRING` to empty string, to be able to deploy as before. If they don't, the startup will fail when the config is substituted. I would suggest an empty default for both env's. Then both options are available. Imho a validation is not necessary. For me it doesn't matter if the service fails on reading the config or when connecting to the database. I assume both errors lead you to the cause in the logs.

Suggestion:

```yaml
mongo:
  connectionString: "${MONGODB_CONNECTION_STRING:-}"
    hosts: "${MONGODB_HOSTS:-}"
    database: "${MONGODB_DATABASE:-}"
    options: "${MONGODB_OPTIONS:-}"
```

Reply:
I changed the documented configuration using empty defaults for `hosts` and `connectionString`.

Reply:
But hosts and database are still required to be `@NotEmpty` in the MongoConfiguration class. I would prefer to remove that and either add some manual validation in the MongoConfiguration class or leave it out completely as Jörg mentioned as well. That would be maximum backwards compatibility, wouldn’t it?